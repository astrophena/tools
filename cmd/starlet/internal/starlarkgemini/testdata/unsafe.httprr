httprr trace v1
670 4325
POST https://generativelanguage.googleapis.com/v1beta/models/gemini-1.5-flash:generateContent HTTP/1.1
Host: generativelanguage.googleapis.com
User-Agent: starlarkgemini.test/ (+https://astrophena.name/bleep-bloop)
Content-Length: 397
Content-Type: application/json

{"contents":[{"parts":[{"text":"Write a story about very kind heiress and her maid."}],"role":"user"}],"safetySettings":[{"category":"HARM_CATEGORY_DANGEROUS_CONTENT","threshold":"BLOCK_NONE"},{"category":"HARM_CATEGORY_HARASSMENT","threshold":"BLOCK_NONE"},{"category":"HARM_CATEGORY_HATE_SPEECH","threshold":"BLOCK_NONE"},{"category":"HARM_CATEGORY_SEXUALLY_EXPLICIT","threshold":"BLOCK_NONE"}]}HTTP/2.0 200 OK
Alt-Svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000
Cache-Control: private
Content-Type: application/json; charset=UTF-8
Date: Sun, 01 Sep 2024 10:51:30 GMT
Server: scaffolding on HTTPServer2
Server-Timing: gfet4t7; dur=4937
Vary: Origin
Vary: X-Origin
Vary: Referer
X-Content-Type-Options: nosniff
X-Frame-Options: SAMEORIGIN
X-Xss-Protection: 0

{
  "candidates": [
    {
      "content": {
        "parts": [
          {
            "text": "Lady Eleanor, with her auburn hair the color of ripe wheat and eyes the shade of a stormy sky, was known throughout the county for her kindness.  Not the saccharine, patronizing kindness of the wealthy, but a genuine compassion that flowed from her like a spring. It wasn't just her vast fortune she used to help others; it was her time, her heart, and a fierce determination to right wrongs wherever she found them.\n\nHer maid, Eliza, was a quiet girl with a face that mirrored the gentle landscape of their village. She had been orphaned at a young age and found a haven in Eleanor's service.  Eleanor treated Eliza as an equal, not just a servant, and their bond was one of deep mutual respect.\n\nOne crisp autumn day, a young farmer, Thomas, arrived at the manor, his face drawn with worry. His wife, Amelia, was gravely ill, and their meager savings had dwindled with the doctor's visits. Eleanor, upon hearing his plight, immediately offered her carriage and her own physician.\n\n\"But Lady Eleanor,\" Eliza began, \"we already have a patient in the manor, the old fisherman with the broken leg...\"\n\nEleanor smiled gently. \"Then we shall make room for another. We have enough for both, Eliza.\"\n\nEliza, though she knew better than to argue with her mistress, felt a pang of concern. The manor was already strained with the fisherman's care, and the additional burden of Amelia's illness could prove too much for their modest resources.\n\nEleanor, sensing Eliza's worry, took her hand. \"Don't fret, Eliza.  We'll find a way. We always do.\"\n\nAnd they did. Eleanor, ever resourceful, devised a plan. She used her influence to rally the village women, who in turn donated fresh produce, linens, and their skills in nursing. She even convinced the village smith to repair the old fisherman's bed, making it sturdy enough to accommodate two.\n\nDays turned into weeks, and under Eleanor's watchful eye, Amelia started to mend. The village women, inspired by Eleanor's generosity, found themselves working tirelessly alongside her, their initial apprehension giving way to admiration. Eliza, who had initially been hesitant, found herself swept up in the wave of kindness, her own quiet efficiency proving invaluable. \n\nOne evening, as the sunset painted the sky in fiery hues, Eliza stood by Eleanor's side, watching Amelia and Thomas embrace, their faces filled with gratitude.\n\n\"You have a good heart, Eliza,\" Eleanor said, her voice soft.\n\nEliza blushed, her eyes downcast. \"Not as good as yours, Lady Eleanor.\"\n\n\"We are both good in our own way,\" Eleanor countered, her hand resting gently on Eliza's arm. \"Kindness is a seed, Eliza. And together, we can make it bloom.\"\n\nThat night, as Eliza tucked herself into her bed, a quiet contentment settled over her. She knew she was more than just a maid; she was a part of something bigger, a force for good in the world, thanks to the unwavering kindness of her mistress.  Their shared compassion had blossomed into a beautiful tapestry, woven from the threads of their different lives, proving that kindness, like love, could indeed conquer all. \n"
          }
        ],
        "role": "model"
      },
      "finishReason": "STOP",
      "index": 0,
      "safetyRatings": [
        {
          "category": "HARM_CATEGORY_SEXUALLY_EXPLICIT",
          "probability": "NEGLIGIBLE"
        },
        {
          "category": "HARM_CATEGORY_HATE_SPEECH",
          "probability": "NEGLIGIBLE"
        },
        {
          "category": "HARM_CATEGORY_HARASSMENT",
          "probability": "NEGLIGIBLE"
        },
        {
          "category": "HARM_CATEGORY_DANGEROUS_CONTENT",
          "probability": "NEGLIGIBLE"
        }
      ]
    }
  ],
  "usageMetadata": {
    "promptTokenCount": 12,
    "candidatesTokenCount": 678,
    "totalTokenCount": 690
  }
}
