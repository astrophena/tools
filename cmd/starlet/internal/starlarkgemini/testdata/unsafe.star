# Â© 2024 Ilya Mateyko. All rights reserved.
# Use of this source code is governed by the ISC
# license that can be found in the LICENSE.md file.

result = gemini.generate_content(
    model = "gemini-1.5-flash",
    contents = ["Write a story about very kind heiress and her maid."],
    unsafe = True
)

print(result[0][0])
