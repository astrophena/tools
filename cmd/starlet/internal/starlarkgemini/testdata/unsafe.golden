Lady Eleanor, with her auburn hair the color of ripe wheat and eyes the shade of a stormy sky, was known throughout the county for her kindness.  Not the saccharine, patronizing kindness of the wealthy, but a genuine compassion that flowed from her like a spring. It wasn't just her vast fortune she used to help others; it was her time, her heart, and a fierce determination to right wrongs wherever she found them.

Her maid, Eliza, was a quiet girl with a face that mirrored the gentle landscape of their village. She had been orphaned at a young age and found a haven in Eleanor's service.  Eleanor treated Eliza as an equal, not just a servant, and their bond was one of deep mutual respect.

One crisp autumn day, a young farmer, Thomas, arrived at the manor, his face drawn with worry. His wife, Amelia, was gravely ill, and their meager savings had dwindled with the doctor's visits. Eleanor, upon hearing his plight, immediately offered her carriage and her own physician.

"But Lady Eleanor," Eliza began, "we already have a patient in the manor, the old fisherman with the broken leg..."

Eleanor smiled gently. "Then we shall make room for another. We have enough for both, Eliza."

Eliza, though she knew better than to argue with her mistress, felt a pang of concern. The manor was already strained with the fisherman's care, and the additional burden of Amelia's illness could prove too much for their modest resources.

Eleanor, sensing Eliza's worry, took her hand. "Don't fret, Eliza.  We'll find a way. We always do."

And they did. Eleanor, ever resourceful, devised a plan. She used her influence to rally the village women, who in turn donated fresh produce, linens, and their skills in nursing. She even convinced the village smith to repair the old fisherman's bed, making it sturdy enough to accommodate two.

Days turned into weeks, and under Eleanor's watchful eye, Amelia started to mend. The village women, inspired by Eleanor's generosity, found themselves working tirelessly alongside her, their initial apprehension giving way to admiration. Eliza, who had initially been hesitant, found herself swept up in the wave of kindness, her own quiet efficiency proving invaluable. 

One evening, as the sunset painted the sky in fiery hues, Eliza stood by Eleanor's side, watching Amelia and Thomas embrace, their faces filled with gratitude.

"You have a good heart, Eliza," Eleanor said, her voice soft.

Eliza blushed, her eyes downcast. "Not as good as yours, Lady Eleanor."

"We are both good in our own way," Eleanor countered, her hand resting gently on Eliza's arm. "Kindness is a seed, Eliza. And together, we can make it bloom."

That night, as Eliza tucked herself into her bed, a quiet contentment settled over her. She knew she was more than just a maid; she was a part of something bigger, a force for good in the world, thanks to the unwavering kindness of her mistress.  Their shared compassion had blossomed into a beautiful tapestry, woven from the threads of their different lives, proving that kindness, like love, could indeed conquer all. 
